src/v/kafka/types.h

```diff
@@ -119,6 +119,32 @@ inline kafka::leader_epoch leader_epoch_from_term(model::term_id term) {
         return kafka::invalid_leader_epoch;
     }
 }
+
+inline std::ostream& operator<<(std::ostream& os, describe_configs_type t) {
+    switch (t) {
+    case describe_configs_type::unknown:
+        return os << "{unknown}";
+    case describe_configs_type::boolean:
+        return os << "{boolean}";
+    case describe_configs_type::string:
+        return os << "{string}";
+    case describe_configs_type::_int:
+        return os << "{int}";
+    case describe_configs_type::_short:
+        return os << "{short}";
+    case describe_configs_type::_long:
+        return os << "{long}";
+    case describe_configs_type::_double:
+        return os << "{double}";
+    case describe_configs_type::list:
+        return os << "{list}";
+    case describe_configs_type::_class:
+        return os << "{class}";
+    case describe_configs_type::password:
+        return os << "{password}";
+    }
+    return os << "{unknown}";
+}
```

Comment:
please move into a `.cc` file out of the header. it'll help with compilation times and speed.